src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
         }
         Entry<K, V> evicted = head;
         head = evicted.getNextEntry();
-        head.setPreEntry(null);
+        if (head != null) {
+            head.setPreEntry(null);
+        }
         evicted.setNextEntry(null);
+        
```

Comment:
**Scalability Issue**: Removing the loop that handles reducing the cache size to the new capacity upon resizing can lead to a cache size exceeding the specified capacity, negatively impacting memory scalability. <br> **Fix**: Reinstate the loop that iterates over the cache to evict entries until the cache size is equal to the new capacity, ensuring the cache does not exceed its maximum capacity. <br> **Code Suggestion**: 
 ```
 +        for (int i = data.size(); i > newCapacity; i--) {
 +            Entry<K, V> evicted = evict();
 +            data.remove(evicted.getKey());
 +        }
 ```